presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryColumnWriter.java

```diff
@@ -280,6 +284,7 @@ public OptionalInt tryConvertToDirect(int maxDirectBytes)
         // Conversion to DirectStream succeeded, Transfer the present stream to direct writer and assign
         // this a new PresentStream, so one writer is responsible for one present stream.
         movePresentStreamToDirectWriter(presentStream);
+        updateRawSizeInDirectWriter(rawSize);
```

Comment:
Dictionary writer continuously attempts to write the data as direct encoded to see if it makes sense to keep the dictionary encoding. When it so it passes the present stream into the direct writer and then writes values and not blocks(!) directly(!) into the direct column writer. And when it writes data directly, it prevents the direct writer from calculating raw size. To mitigate it we accumulate row group raw sizes in the dictionary writer and then send them to the direct writer for each row group.